cookie-store/resources/cookie_helper.py

```diff
@@ -18,52 +18,68 @@
 #   Used to set cookies from an HTTP response header context.
 #
 # The response has 200 status and content-type: text/plain; charset=<charset>
-import cgi, encodings, os, re, sys, urllib
+import encodings, re
+
+from six import PY3
+
+from six.moves.urllib.parse import parse_qs, quote
+
+from wptserve.utils import isomorphic_decode, isomorphic_encode
 
 # NOTE: These are intentionally very lax to permit testing
-DISALLOWED_IN_COOKIE_NAME_RE = re.compile(r'[;\0-\x1f\x7f]');
-DISALLOWED_IN_HEADER_RE = re.compile(r'[\0-\x1f\x7f]');
+DISALLOWED_IN_COOKIE_NAME_RE = re.compile(br'[;\0-\x1f\x7f]')
+DISALLOWED_IN_HEADER_RE = re.compile(br'[\0-\x1f\x7f]')
 
 # Ensure common charset names do not end up with different
 # capitalization or punctuation
 CHARSET_OVERRIDES = {
     encodings.codecs.lookup(charset).name: charset
-    for charset in ('utf-8', 'iso-8859-1',)
+    for charset in (u'utf-8', u'iso-8859-1',)
 }
 
 def main(request, response):
   assert request.method in (
-      'GET',
-      'POST',
-  ), 'request method was neither GET nor POST: %r' % request.method
-  qd = (request.url.split('#')[0].split('?', 1) + [''])[1]
-  if request.method == 'POST':
-    qd += '&' + request.body
-  args = cgi.parse_qs(qd, keep_blank_values = True)
-  charset = encodings.codecs.lookup(args.get('charset', ['utf-8'])[-1]).name
+      u'GET',
+      u'POST',
+  ), u'request method was neither GET nor POST: %r' % request.method
+  qd = (isomorphic_encode(request.url).split(b'#')[0].split(b'?', 1) + [b''])[1]
+  if request.method == u'POST':
+    qd += b'&' + request.body
+  if PY3:
+    args = parse_qs(qd.decode("iso-8859-1"), keep_blank_values=True, encoding='iso-8859-1')
+  else:
+    args = parse_qs(qd, keep_blank_values=True)
+  charset = encodings.codecs.lookup(args.get(u'charset', [u'utf-8'])[-1]).name
   charset = CHARSET_OVERRIDES.get(charset, charset)
-  headers = [('content-type', 'text/plain; charset=' + charset)]
+  headers = [(b'content-type', b'text/plain; charset=' + isomorphic_encode(charset))]
   body = []
-  if request.method == 'POST':
+  if request.method == u'POST':
     for set_cookie in args.get('set-cookie', []):
       if '=' in set_cookie.split(';', 1)[0]:
         name, rest = set_cookie.split('=', 1)
         assert re.search(
             DISALLOWED_IN_COOKIE_NAME_RE,
-            name
+            isomorphic_encode(name)
         ) is None, 'name had disallowed characters: %r' % name
       else:
         rest = set_cookie
       assert re.search(
           DISALLOWED_IN_HEADER_RE,
-          rest
+          isomorphic_encode(rest)
       ) is None, 'rest had disallowed characters: %r' % rest
-      headers.append(('set-cookie', set_cookie))
-      body.append('set-cookie=' + urllib.quote(set_cookie, ''))
+      headers.append((b'set-cookie', isomorphic_encode(set_cookie)))
+      if PY3:
```

Comment:
It would be nice to factor out a local wrapper for `six.moves.urllib.parse.quote()` that encompases the `if PY3 / else` logic, to keep this surrounding code more readable.

Reply:
Maybe the same for `parse_qs` above, even though that only occurs once?

Reply:
I agree we should create a wrapper for `quote`. See my other comment w.r.t. `parse_qs`.

Reply:
Code updated. Thank you!